framework_tvb/tvb/core/neotraits/forms.py

```diff
@@ -412,6 +394,11 @@ def _from_post(self):
         else:
             self.data = None
 
+    @property
+    def value(self):
+        if self.data == 0:
+            return self.data
+        return super(FloatField, self).value
```

Comment:
I suppose this is the fix (plus the `SimpleFloatField .value` implementation).

It looks a bit fishy indeed. I conquer that a review would be necessary, to check why and how `value` has ended up as string instead of float. As we have the review task already in mind, I think it is acceptable if we merge these changes now (with a TODO maybe referring to the refactor task number) to avoid the bug, and we come back to it when the refactor is to be done.
It would be great if a unit-test on this mysterious `0` value is added (either now or when the refactoring task will be implemented).

Reply:
The 'how' is as follows: 
- we have `data` and `unvalidated_data` attributes on `Field`
- `unvalidated_data` is what comes from the GUI and gets through validation/transformation
- `data` is the processed value (eg. float value for `FloatField`)
- on `Field.value` we have: `return self.data or self.unvalidated_data`

In the current bug, we have the following values:
- data = 0
- unvalidated_data = '0'

So, `self.data or self.unvalidated_data `evaluates to '0' and this is being returned to the visualizer.

We should review the usage of 'data' and 'unvalidated_data' and better define their roles in order to correct such edge cases. But I think it should be done as a separate task in order to better focus on them.

Reply:
Thanks for documenting how this is happening!
Then to me `self.data or self.unvalidated_data` expression seems inappropriate (and maybe not only for Float).
If the role remains as you suggested (`unvalidated_data` comes from GUI, and `data` is after processing), then I suggest to return on property `value` `self.data` by default (unless otherwise overwritten).
But indeed, this requires a deeper validation and it should be in the separate task.